models/cam-fv/shell_scripts/cesm2_1/setup_pmo

```diff
@@ -0,0 +1,1277 @@
+      endif
+
+      ${LINK} ${stagedir}/${refcase}.clm2\${true_string}.r.${init_time}.nc \
+                          ${refcase}.clm2\${inst_string}.r.${init_time}.nc
+
+      ${LINK} ${stagedir}/${refcase}.cice\${true_string}.r.${init_time}.nc \
+                          ${refcase}.cice\${inst_string}.r.${init_time}.nc
+
+      ${LINK} ${stagedir}/${refcase}.cam\${true_string}.i.${init_time}.nc  \
+                             cam_initial\${inst_string}.nc
+
+      # If you are using the river runoff model, you must specify an initial file here
+      if (${COMP_ROF} == 'rtm') then
+         ${LINK} -f ${stagedir}/${refcase}.rtm\${true_string}.r.${init_time}.nc  \
+                             ${refcase}.rtm\${inst_string}.r.${init_time}.nc
+      else if (${COMP_ROF} == 'mosart') then
+         ${LINK} -f ${stagedir}/${refcase}.mosart\${true_string}.r.${init_time}.nc  \
+                             ${refcase}.mosart\${inst_string}.r.${init_time}.nc
+      endif
+
+      @ inst ++
+   end
+
+   echo "All files set to run the FIRST experiment step using (ref)time" $init_time
+
+endif
+exit 0
+
+EndOfText
+chmod 0755 stage_cesm_files
+
+./stage_cesm_files
+
+# ==============================================================================
+# Build the case
+# ==============================================================================
+
+echo ''
+echo 'Building the case'
+echo "   --skip-provenance-check is used.  See note in $0"
+echo ''
+
+# --skip-provenance-check because of svn or git timing out during build
+# of CLM.  It wanted authentication(?) to access a private repository.
+# A better solution would be to find out why(whether) it thinks CLM is
+# a private repository.
+
+./case.build --skip-provenance-check
+
+if ( $status != 0 ) then
+   echo "ERROR: Case could not be built."
+   exit 120
+endif
+
+# ==============================================================================
+# Check some XML settings which should have been set by this script.
+# ==============================================================================
+
+echo ' '
+echo 'CESM settings which are of special interest:'
+echo ' '
+foreach var ( CONTINUE_RUN RESUBMIT RUN_REF RUN_STARTDATE STOP_OPTION \
+             STOP_N SSTICE DOUT RUNDIR MPI_RUN_COMMAND AVGHIST ASSIM)
+   ./xmlquery --partial $var  | grep -v 'Results in' | grep -v '^$'
+end
+echo ' '
+
+# ==============================================================================
+# What to do next
+# ==============================================================================
+
+cat << EndOfText >! CESM_instructions.txt
+
+-------------------------------------------------------------------------
+Time to check the case.
+
+1)  Scan the output from this setup script for errors and warnings:
+    ERROR, WARNING, 'No such file' (except for MOSART)
+    'File status unknown' can be ignored.
+    'ERROR: cice.buildlib failed' can be ignored, unless you've changed the CICE code
+
+2)  cd ${RUNDIR}
```

Comment:
`${RUNDIR}` does not seem to print the correct value:


Here is the output

```
CESM settings which are of special interest:
 
        CONTINUE_RUN: FALSE
        RESUBMIT_SETS_CONTINUE_RUN: TRUE
        RESUBMIT: 0
        RESUBMIT_SETS_CONTINUE_RUN: TRUE
        RUN_REFCASE: cesm2_1_relsd_m5.6
        RUN_REFDATE: 2016-12-10
        RUN_REFDIR: /glade/work/raeder/Models/CAM_init/FV1deg_cesm2_1/2016-12-10-00000
        RUN_REFTOD: 00000
        RUN_STARTDATE: 2009-08-02
        STOP_OPTION: nhours
        STOP_N: 6
        SSTICE_GRID_FILENAME: /glade/p/cesm/cseg/inputdata/share/domains/domain.ocn.fv0.9x1.25_gx1v7.151020.nc
        SSTICE_YEAR_ALIGN: 1850
        SSTICE_YEAR_END: 2021
        SSTICE_YEAR_START: 1850
        SSTICE_DATA_FILENAME: /glade/p/cesm/cseg/inputdata/atm/cam/sst/sst_HadOIBl_bc_0.9x1.25_1850_2020_c210521.nc
        SSTICE_STREAM: CAMDATA
        DOUT_S: FALSE
        DOUT_S_SAVE_INTERIM_RESTART_FILES: FALSE
        DOUT_S_ROOT: /glade/scratch/hkershaw/Kevin_test/Test0/archive
        RUNDIR: /glade/scratch/hkershaw/Kevin_test/Test0/run
        MPI_RUN_COMMAND: UNSET
        AVGHIST_DATE: -999
        AVGHIST_N: -999
        AVGHIST_OPTION: never
        DATA_ASSIMILATION: ['CPL:FALSE', 'ATM:FALSE', 'LND:FALSE', 'ICE:FALSE', 'OCN:FALSE', 'ROF:FALSE', 'GLC:FALSE', 'WAV:FALSE']
        DATA_ASSIMILATION_CYCLES: 1
        DATA_ASSIMILATION_SCRIPT: 
 

-------------------------------------------------------------------------
Time to check the case.

1)  Scan the output from this setup script for errors and warnings:
    ERROR, WARNING, 'No such file' (except for MOSART)
    'File status unknown' can be ignored.
    'ERROR: cice.buildlib failed' can be ignored, unless you've changed the CICE code

2)  cd /glade/scratch/hkershaw/Test0/run
    Check the files that were staged; follow the links to confirm the data sources.
    Check the compatibility between them and the namelists and pointer files.

3)  cd /glade/work/hkershaw/Exp/Test0
    Verify the CESM XML settings, especially in env_batch.xml and env_run.xml.
    ./xmlquery --partial <partial_string_of_interest> 
    is particularly useful.
    Check the contents of the user_nl_{component}* files (e.g. do CAM's external 
    forcing files cover your time period)

4)  The default initial configuration is to do no harvesting of observations.
    as defined by the 'no_assimilate.csh' in the env_run.xml file.
    When you are ready to harvest, change "no_assimilate" to "perfect_model"
    and follow the instructions in 
    /glade/work/hkershaw/Exp/Test0/DART_config. 
```

The rundir from cesm is  `/glade/scratch/hkershaw/Kevin_test/Test0/run`  not `/glade/scratch/hkershaw/Test0/run`

Reply:
Good catch.  RUNDIR in setup_pmo is set (to CESM's default value) 
before CESM's CIME_OUTPUT_ROOT is set to the value that you want (with Kevin_test in the name),
which the changes RUNDIR in CESM, but not in setup_pmo.

So the 'setenv RUNDIR' line should be moved to after the 'setenv CIME_OUTPUT_ROOT' line:

> ./xmlchange CIME_OUTPUT_ROOT=${cime_output}
> \# Changing CIME_OUTPUT_ROOT changes RUNDIR in CESM, so change it here too.
> setenv RUNDIR  `./xmlquery RUNDIR --value`

I'll commit that change (and in setup_hybrid and setup_advanced).